obspy/io/hypodd/pha.py

```diff
@@ -111,19 +128,128 @@ def _read_pha(filename, inventory=None, id_map=None, id_default='.{}..{}',
         substituted by station code and component.
     :param dict ph2comp: mapping of phases to components
         (default: {'P': 'Z', 'S': 'N'})
+    :param dict eventid_map: Desired mapping of hypodd event ids (dict values)
+        to event resource ids (dict keys).
+        The returned dictionary of the HYPODDPHA writing operation can be used.
+        By default, ids are not mapped.
     :param str encoding: encoding used (default: utf-8)
 
     :rtype: :class:`~obspy.core.event.Catalog`
     :return: An ObsPy Catalog object.
     """
+    if eventid_map is not None:
+        eventid_map = {v: k for k, v in eventid_map.items()}
     seed_map = _seed_id_map(inventory, id_map)
     with io.open(filename, 'r', encoding=encoding) as f:
         text = f.read()
-    events = [_block2event(block, seed_map, id_default, ph2comp)
+    events = [_block2event(block, seed_map, id_default, ph2comp, eventid_map)
               for block in text.split('#')[1:]]
     return Catalog(events)
 
 
+PHA1 = ('# {o.time.year} {o.time.month:>2} {o.time.day:>2} {o.time.hour:>2}'
+        ' {o.time.minute:>2} {o.time.second:>2}.{o.time.microsecond:06d}  '
+        '{o.latitude} {o.longitude} {depth}  {mag}  {he} {ve} {rms}'
+        '   {evid:>9}\n')
+PHA2 = '{p.waveform_id.station_code:6}  {relt:.4f}  {weight}  {p.phase_hint}\n'
+
+
+def _map_eventid(evid, eventid_map, used_ids, counter):
```

Comment:
is `used_ids` always going to `set(eventid_map.values())`? If so, it might be worth removing `used_ids` as an argument.

Reply:
It's not, because the eventid_map is only populated if idpha and eventid are different.